src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_PointerElementAccess.cs

```diff
@@ -12,12 +8,24 @@ internal sealed partial class LocalRewriter
     {
         public override BoundNode VisitPointerElementAccess(BoundPointerElementAccess node)
         {
-            BoundExpression rewrittenExpression = VisitExpression(node.Expression);
+            BoundExpression rewrittenExpression = LowerReceiverOfPointerElementAccess(node.Expression);
             BoundExpression rewrittenIndex = VisitExpression(node.Index);
 
             return RewritePointerElementAccess(node, rewrittenExpression, rewrittenIndex);
         }
 
+        private BoundExpression LowerReceiverOfPointerElementAccess(BoundExpression receiver)
+        {
+            if (receiver is BoundFieldAccess fieldAccess && fieldAccess.FieldSymbol.IsFixed)
+            {
+                var loweredFieldReceiver = VisitExpression(fieldAccess.ReceiverOpt);
+                fieldAccess.Update(loweredFieldReceiver, fieldAccess.FieldSymbol, fieldAccess.ConstantValueOpt, fieldAccess.ResultKind, fieldAccess.Type);
+                return new BoundAddressOfOperator(receiver.Syntax, fieldAccess, isManaged: true, fieldAccess.Type);
```

Comment:
>fieldAccess.Type [](start = 97, length = 16)

Is this type gong to be a pointer type?

Reply:
Yes. A pointer. 
Technically it should be "managed pointer", but we do not have a way to express it and I did not feel like it is needed just for this case. 
There will be no much distinction since both managed and unmanaged pointers can be indexed.

---
In reply to: [170481112](https://github.com/dotnet/roslyn/pull/24966#discussion_r170481112) [](ancestors = 170481112)